src/dbnode/persist/fs/wide/index_checksum_block_buffer.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE
+
+package wide
+
+import (
+	"github.com/m3db/m3/src/x/ident"
+
+	"go.uber.org/atomic"
+)
+
+type indexChecksumBlockBuffer struct {
+	closed       *atomic.Bool
+	currentBlock ident.IndexChecksumBlock
+	blocks       chan ident.IndexChecksumBlock
+	buffer       []ident.IndexChecksumBlock
+}
+
+// NewIndexChecksumBlockBuffer creates a new IndexChecksumBlockBuffer.
+func NewIndexChecksumBlockBuffer(
+	blockInput chan ident.IndexChecksumBlock,
+) IndexChecksumBlockBuffer {
+	return &indexChecksumBlockBuffer{
+		closed: atomic.NewBool(false),
+		blocks: blockInput,
+		buffer: make([]ident.IndexChecksumBlock, 0, 10),
+	}
+}
+
+func (b *indexChecksumBlockBuffer) Close() {
+	if !b.closed.Swap(true) {
+		close(b.blocks)
```

Comment:
Is the receiver supposed to close this channel? I believe the general recommendation is for the sender to do it, besides, that's how you do it in the test.

Reply:
Good call; this has no need to close as of yet, so will remove this function